Yoast/wordpress-seo
# 19760 restructuring yoastseo correctly bundle the word complexity assessment

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Moves the registration of the _word complexity_ assessment and its research and helper outside of `yoastseo`.
* [yoastseo] Removes the logic to initialize the _word complexity_ assessment from the worker and the app.
* [yoastseo] Adds additional logic to register a readability assessment for cornerstone content.

## Relevant technical choices:

* The obsolete logic for registering Word complexity assessment is also removed in this PR. This is done to make sure that the new way of registering the assessment works properly.
* The word complexity config is not yet registered from premium/shopify in this PR, and will be tackled in a different issue. This is because `wordComplexity` config is a config in which we don't have an interface in the worker to register a config to the researcher. In the researcher level, we have a way of registering data `.addResearchData()`. Alternatively we could also register the `wordComplexity` config as a research data. However, the only worker interface to register research data is upon worker initialization:
````
YoastSEO.analysis.worker.initialize( {
	researchData: {
		morphology: morphologyData,
	},
} )
````

* Having the above point in mind and also the fact that this PR's scope is already big, it's decided that tackling the problem of registering the assessment config in a different issue is better. The issue [Restructuring yoastseo: bundle correctly word complexity assessment's config](https://github.com/Yoast/wordpress-seo/issues/19822)


## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

The accompanying PRs:
* `wordpress-seo-premium`: https://github.com/Yoast/wordpress-seo-premium/pull/3807
* `shopify-seo`: https://github.com/Yoast/shopify-seo/pull/1021


Testing steps:
* Execute the testing steps in these scenarios: 
   * [Word complexity assessment](https://docs.google.com/document/d/1fWX_8YkvSqcsqmwan_gwJjYqGOKIcTBTv0FbLKz47ig/edit#heading=h.adgajq24246l)
   * [Bundled word complexity code when premium is not activated](https://docs.google.com/document/d/1fWX_8YkvSqcsqmwan_gwJjYqGOKIcTBTv0FbLKz47ig/edit#heading=h.y6pguwedewby). You can skip the steps for checking the `wordComplexity` config and the frequency list.
   * Downgrade/upgrade flows:
      * [New Free + Old Premium](https://docs.google.com/document/d/1fWX_8YkvSqcsqmwan_gwJjYqGOKIcTBTv0FbLKz47ig/edit#heading=h.28bzzx4fzuj2) 
      * [Old Free + New Premium](https://docs.google.com/document/d/1fWX_8YkvSqcsqmwan_gwJjYqGOKIcTBTv0FbLKz47ig/edit#heading=h.dqd4s8tizim)
* Additional steps:
   When testing the assessment in one of the supported languages:
   *  Open a post with the word complexity assessment returning an orange bullet
   * Switch on the cornerstone content toggle
   * Confirm that the bullet is turned into red
   * Save the post
   * Reload the post
   * Confirm that the word complexity assessment still returns a red bullet
   * Switch off the cornerstone content toggle
   * Confirm that the bullet is turned into orange
   * Save the post
   * Reload the post
   * Confirm that the word complexity assessment still returns an orange bullet
* Also test in Shopify:
   * Fully test in one content type, smoke test in others 

#### Relevant test scenarios
* [x] Changes should be tested on different posts/pages/taxonomies/custom post types/custom taxonomies/products


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* This PR also touches the worker interface to add an assessment to the assessor (especially readability assessment). But I think it's also good to smoke test the additional assessments registered via the aforementioned worker interface. Therefore, please smoke test some of these assessments mentioned in [this overview](https://yoast.atlassian.net/wiki/spaces/LIN/pages/2618589221/Registering+additional+assessments#Overview-of-our-current-additional-assessments).

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes https://github.com/Yoast/wordpress-seo/issues/19760
